payments_service/braintree/views.py

```diff
@@ -59,6 +60,25 @@ def post(self, request):
         if not form.is_valid():
             return error_400(response=form.errors)
 
+        if settings.ALLOW_REPEAT_PAYMENTS:
+            log.warn('not protecting against repeat subscriptions')
+        else:
+            product = self.api.generic.product.get_object_or_404(
+                public_id=form.cleaned_data['plan_id'],
+            )
+            # Check if the user is already subscribed to this plan.
+            result = self.api.braintree.mozilla.subscription.get(
+                paymethod__braintree_buyer__buyer=self.request.user.pk,
+                seller_product=product['resource_pk'],
+            )
+            if len(result):
+                log.info(
+                    'buyer {buyer} is already subscribed to product {product}'
+                    .format(buyer=self.request.user.pk,
+                            product=product['resource_pk']))
+                return error_400(
+                    response='user is already subscribed to this product')
```

Comment:
I don't think the UI will ever encounter this in normal operation. If it did, it would mean the vendor asked the user to buy something they already own which they shouldn't do.

However, we don't have fancy error codes yet in the service. We should add them! https://github.com/mozilla/payments-service/issues/50